alerting/src/test/kotlin/org/opensearch/alerting/util/destinationmigration/DestinationMigrationUtilServiceIT.kt

```diff
@@ -0,0 +1,105 @@
+import org.opensearch.alerting.util.DestinationType
+import org.opensearch.client.ResponseException
+import org.opensearch.rest.RestStatus
+import java.time.Instant
+import java.util.UUID
+
+class DestinationMigrationUtilServiceIT : AlertingRestTestCase() {
+
+    fun `test migrateData`() {
+        if (isNotificationPluginInstalled()) {
+            // Create alerting config index
+            createRandomMonitor()
+
+            val emailAccount = EmailAccount(
+                name = "test",
+                email = "test@email.com",
+                host = "smtp.com",
+                port = 25,
+                method = EmailAccount.MethodType.NONE,
+                username = null,
+                password = null
+            )
+            val emailAccountDoc = "{\"email_account\" : ${emailAccount.toJsonString()}}"
+            val emailGroup = EmailGroup(
+                name = "test",
+                emails = listOf(EmailEntry("test@email.com"))
+            )
+            val emailGroupDoc = "{\"email_group\" : ${emailGroup.toJsonString()}}"
+            val emailAccountId = UUID.randomUUID().toString()
+            val emailGroupId = UUID.randomUUID().toString()
+            indexDoc(SCHEDULED_JOBS_INDEX, emailAccountId, emailAccountDoc)
+            indexDoc(SCHEDULED_JOBS_INDEX, emailGroupId, emailGroupDoc)
+
+            val recipient = Recipient(Recipient.RecipientType.EMAIL, null, "test@email.com")
+            val email = Email(emailAccountId, listOf(recipient))
+            val emailDest = Destination(
+                id = UUID.randomUUID().toString(),
+                type = DestinationType.EMAIL,
+                name = "test",
+                user = randomUser(),
+                lastUpdateTime = Instant.now(),
+                chime = null,
+                slack = null,
+                customWebhook = null,
+                email = email
+            )
+            val slackDestination = getSlackDestination().copy(id = UUID.randomUUID().toString())
+            val chimeDestination = getChimeDestination().copy(id = UUID.randomUUID().toString())
+            val customWebhookDestination = getCustomWebhookDestination().copy(id = UUID.randomUUID().toString())
+
+            val destinations = listOf(emailDest, slackDestination, chimeDestination, customWebhookDestination)
+
+            val ids = mutableListOf(emailAccountId, emailGroupId)
+            for (destination in destinations) {
+                val dest = """
+                    {
+                      "destination" : ${destination.toJsonString()}
+                    }
+                """.trimIndent()
+                indexDoc(SCHEDULED_JOBS_INDEX, destination.id, dest)
+                ids.add(destination.id)
+            }
+
+            // Create cluster change event and wait for migration service to complete migrating data over
+            client().updateSettings("indices.recovery.max_bytes_per_sec", "40mb")
+            Thread.sleep(120000)
```

Comment:
Is there a way to reduce this time without making it flaky?

Reply:
Hmm, the job will wait 1 minute before actually running for the first time. So if we think once running it doesn't take too long, we might be able to reduce it to as little over 1 minute as possible.

I won't treat it as a blocker for merging if that's fine with you but we can follow-up on it during general testing.

Reply:
sure thats fine